### PR TITLE
Allow extensibility through using a custom GalleryItem

### PR DESCRIPTION
**Pull request by zfoltin**
Using the `.custom` GalleryItem type, people will be able to use their `UIImageView` subclasses in the gallery

Example in the example project. Feedback is welcome

PS: hopefully helps with https://github.com/MailOnline/ImageViewer/issues/56

### CODE REVIEW

File: Example/ViewController.swift [hunk 1]
```diff
@@ -77,6 +77,17 @@ class ViewController: UIViewController, GalleryItemsDatasource, GalleryDisplaced
...
+
+            let myFetchImageBlock: FetchImageBlock = { $0(UIImage(named: "8")!) }
+
+            let getViewController: GetViewControllerCompletion = { index, itemCount, fetchImageBlock, configuration, isInitialController in
+
+                return AnimatedViewController(index: index, itemCount: itemCount, fetchImageBlock: myFetchImageBlock, configuration: configuration, isInitialController: isInitialController)
+            }
+
+            return GalleryItem.custom(fetchImageBlock: myFetchImageBlock, getViewController: getViewController)
+        }
```
Comment (Krisiacik) [inline Example/ViewController.swift]: I would rename the argument name to itemViewController

Comment (zfoltin) [inline Example/ViewController.swift]: 👍 changed it to `itemViewControllerBlock`

**Comment by zfoltin** [comment]
I made ItemBaseController public so it can be inherited from

Now there is still an issue - in the example the displaced view animations don't work on this item :(

**Comment by Krisiacik** [comment]
They should work...I don't see any reason why not..did you implement the displaced Datasource?

**Comment by zfoltin** [comment]
It was to do with not setting the `delegate` and `displacedViewsDatasource` on GalleryItem.custom view controllers.
Also had to make a few more things public in the process but it seems to be working now. Please 👀

**Comment by Krisiacik** [comment]
Looks good...if it indeed works, let's merge merge merge. And win win win.

**Pull request merged**